File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int p1 = m - 1, p2 = n - 1;
        for (int p = n + m - 1; p >= 0; p--) {
            if (p2 < 0)
                break;
            if (p1 >= 0 && nums1[p1] > nums2[p2]) {
                nums1[p] = nums1[p1];
                p1 -= 1;
            } else {
                nums1[p] = nums2[p2];
                p2 -= 1;
            }
        }
    }
}

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
       
        first, second, ind = m-1, n-1, m + n - 1

        while second >= 0:
            if  first >= 0 and nums1[first] > nums2[second]:
                nums1[ind] = nums1[first]
                first -= 1
            else:
                nums1[ind] = nums2[second]
                second -= 1
            ind -= 1